src/org/traccar/geocoder/JsonGeocoder.java

```diff
@@ -84,6 +90,36 @@ public void onThrowable(Throwable t) {
         });
     }
 
+    @Override
+    public String getAddress(final double latitude, final double longitude) {
```

Comment:
There seems to be quite a bit of duplication. Is there way we can avoid it?

Reply:
It looks like that, I can try, but the problem is that some pieces of code inside callbacks and call other callbacks but similar are linear.

Reply:
The part that needs to be async can be separated into a method or something.